hrm-service/src/case/case.ts

```diff
@@ -5,16 +5,104 @@
  */
 import * as caseDb from './case-data-access';
 import { retrieveCategories } from '../controllers/helpers';
+import { CaseRecordCommon, CaseSectionRecord, Contact, NewCaseRecord } from './case-data-access';
+import { randomUUID } from 'crypto';
 
-type Case = CaseRecord & {
+export const WELL_KNOWN_CASE_SECTION_NAMES: Record<
+  string,
+  { sectionTypeName: string; getSectionSpecificData: (section: any) => any }
+> = {
+  households: { getSectionSpecificData: s => s.household, sectionTypeName: 'household' },
+  perpetrators: { getSectionSpecificData: s => s.perpetrator, sectionTypeName: 'perpetrator' },
+  incidents: { getSectionSpecificData: s => s.incident, sectionTypeName: 'incident' },
+  counsellorNotes: { getSectionSpecificData: s => ({ note: s.note }), sectionTypeName: 'note' },
+  referrals: {
+    getSectionSpecificData: s => {
+      return { date: s.date, referredTo: s.referredTo, comments: s.comments };
+    },
+    sectionTypeName: 'referral',
+  },
+  documents: { getSectionSpecificData: s => s.document, sectionTypeName: 'document' },
+};
+
+export const EMPTY_INFO_SECTIONS = Object.fromEntries(
+  Object.entries(WELL_KNOWN_CASE_SECTION_NAMES).map(([k]) => [k, []]),
+);
+
+export type Case = CaseRecordCommon & {
+  id: number;
   childName?: string;
   categories: Record<string, string[]>;
+  connectedContacts?: Contact[];
 };
 
+type CaseInfoSection = {
+  id: string;
+  twilioWorkerId: string;
+  updatedAt?: string;
+  updatedBy?: string;
+} & Record<string, any>;
+
 type CaseRecord = caseDb.CaseRecord;
 
+/**
+ * Converts a single list of all sections for a case to a set of arrays grouped by type
+ */
+const caseSectionRecordsToInfo = (
+  sections: CaseSectionRecord[] = [],
+): Record<string, CaseInfoSection[]> => {
+  const infoLists: Record<string, CaseInfoSection[]> = {};
+  return sections.reduce((categorized, record) => {
+    const {
+      caseId,
+      sectionType,
+      sectionId,
+      sectionTypeSpecificData,
+      createdBy,
+      ...restOfRecord
+    } = record;
+
+    if (!restOfRecord.updatedAt) {
+      delete restOfRecord.updatedAt;
+    }
+    if (!restOfRecord.updatedBy) {
+      delete restOfRecord.updatedBy;
+    }
```

Comment:
Really just to line up with what the existing tests expected - when attempting to make a '100% compatible refactor' refactor it's best to have an abundance of caution when deciding how to handle minor differences like this. I'd rather have some redundant code and remove it later rather than assume it's redundant from the start and introduce bugs